BoostedAnalyzer/src/MCMatchVarProcessor.cpp

```diff
@@ -383,7 +445,75 @@ bool dfirst=true;
     if(minDrB2<.25){
       vars.FillVar( "GenHiggs_B2_Idx",idxb2);
     }
+
+    // get mass of dijet system of B indices were found
+    if( minDrB1<.25 && minDrB2<.25 )
+    {
+        // get dijet
+        math::XYZTLorentzVector higgs_vec = jetvecs[idxb1]+jetvecs[idxb2];
+        vars.FillVar("GenHiggsMassFromMatchedJets", higgs_vec.M());
+    }
+
   }
+
```

Comment:
Whats happening here?

Reply:
Hello sebwieland,
thank you for your comment.

This part of the module searches for two jets matching the generator level Higgs decay products. If two jets are found it calculates the invariant dijet mass, which can be used to generate a mass template for a mass based reconstruction method, like the implemented Chi2-Reconstruction.